src/Reflection/FieldItemListPropertyReflection.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace PHPStan\Reflection;
+
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+
+/**
+ * Allows field access via magic methods
+ *
+ * See \Drupal\Core\Field\FieldItemListInterface::__get and ::__set.
+ */
+class FieldItemListPropertyReflection implements PropertyReflection
+{
+
+  /** @var ClassReflection */
+    private $declaringClass;
+
+  /** @var string */
+    private $propertyName;
+
+    public function __construct(ClassReflection $declaringClass, string $propertyName)
+    {
+        $this->declaringClass = $declaringClass;
+        $this->propertyName = $propertyName;
+    }
+
+    public static function canHandleProperty(string $propertyName): bool
+    {
+        $names = ['entity', 'value', 'target_id'];
+        return in_array($propertyName, $names, true);
```

Comment:
Should add a comment on "how to handle fields when this isn't true" like Address field.